keras/layers/local.py

```diff
@@ -245,6 +245,7 @@ def build(self, input_shape):
       self.input_spec = InputSpec(ndim=3, axes={1: input_dim})
     else:
       self.input_spec = InputSpec(ndim=3, axes={-1: input_dim})
+    self.op_shape = self.compute_output_shape(input_shape)
```

Comment:
This caching is error-prone, why add it? `build` is only called once but the layer can be called with inputs of various shapes, so the output shape should not be cached

Reply:
I had added this because earlier (when this PR was originally opened under tensorflow) `inputs.shape` returned a bunch of `None` values, which the led to [#47689](https://github.com/tensorflow/tensorflow/issues/47689). Thus, I thought this could be a workaround.  But that doesn't seem to be a problem now, so I've removed it.